FreeRTOS-Plus/Demo/AWS/Fleet_Provisioning_Windows_Simulator/Fleet_Provisioning_With_CSR_Demo/DemoTasks/FleetProvisioningDemoExample.c

```diff
@@ -0,0 +1,831 @@
+    }
+
+    if( xStatus == true )
+    {
+        xStatus = xSubscribeToTopic( FP_CBOR_REGISTER_REJECTED_TOPIC( democonfigPROVISIONING_TEMPLATE_NAME ),
+                                     FP_CBOR_REGISTER_REJECTED_LENGTH( fpdemoPROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+        if( xStatus == false )
+        {
+            LogError( ( "Failed to subscribe to fleet provisioning topic: %.*s.",
+                        FP_CBOR_REGISTER_REJECTED_LENGTH( fpdemoPROVISIONING_TEMPLATE_NAME_LENGTH ),
+                        FP_CBOR_REGISTER_REJECTED_TOPIC( democonfigPROVISIONING_TEMPLATE_NAME ) ) );
+        }
+    }
+
+    return xStatus;
+}
+/*-----------------------------------------------------------*/
+
+static bool prvUnsubscribeFromRegisterThingResponseTopics( void )
+{
+    bool xStatus;
+
+    xStatus = xUnsubscribeFromTopic( FP_CBOR_REGISTER_ACCEPTED_TOPIC( democonfigPROVISIONING_TEMPLATE_NAME ),
+                                     FP_CBOR_REGISTER_ACCEPTED_LENGTH( fpdemoPROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+    if( xStatus == false )
+    {
+        LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                    FP_CBOR_REGISTER_ACCEPTED_LENGTH( fpdemoPROVISIONING_TEMPLATE_NAME_LENGTH ),
+                    FP_CBOR_REGISTER_ACCEPTED_TOPIC( democonfigPROVISIONING_TEMPLATE_NAME ) ) );
+    }
+
+    if( xStatus == true )
+    {
+        xStatus = xUnsubscribeFromTopic( FP_CBOR_REGISTER_REJECTED_TOPIC( democonfigPROVISIONING_TEMPLATE_NAME ),
+                                         FP_CBOR_REGISTER_REJECTED_LENGTH( fpdemoPROVISIONING_TEMPLATE_NAME_LENGTH ) );
+
+        if( xStatus == false )
+        {
+            LogError( ( "Failed to unsubscribe from fleet provisioning topic: %.*s.",
+                        FP_CBOR_REGISTER_REJECTED_LENGTH( fpdemoPROVISIONING_TEMPLATE_NAME_LENGTH ),
+                        FP_CBOR_REGISTER_REJECTED_TOPIC( democonfigPROVISIONING_TEMPLATE_NAME ) ) );
+        }
+    }
+
+    return xStatus;
+}
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Create the task that demonstrates the Fleet Provisioning library API
+ */
+void vStartFleetProvisioningDemo()
+{
+    /* Configure mbedTLS to use FreeRTOS specific threading function. */
+    mbedtls_threading_set_alt( mbedtls_platform_mutex_init,
+                               mbedtls_platform_mutex_free,
+                               mbedtls_platform_mutex_lock,
+                               mbedtls_platform_mutex_unlock );
+
+    /* This example uses a single application task, which shows that how to use
+     * Device Defender library to generate and validate AWS IoT Device Defender
+     * MQTT topics, and use the coreMQTT library to communicate with the AWS
+     * IoT Device Defender service. */
+    xTaskCreate( prvFleetProvisioningTask, /* Function that implements the task. */
+                 "DemoTask",               /* Text name for the task - only used for debugging. */
```

Comment:
Would be helpful to name this something specific to fleet provisioning.

Reply:
The "DemoTask" name is consistent with other demos, and the logs will contain "FLEET_PROVISIONING_DEMO" whenever a something within the solution logs - should it still be renamed?

```
9 8639 [DemoTask] [INFO] [FLEET_PROVISIONING_DEMO] [prvProvisioningPublishCallback:297] Received accepted response from Fleet Provisioning CreateCertificateFromCsr API.
10 9540 [DemoTask] [INFO] [FLEET_PROVISIONING_DEMO] [prvFleetProvisioningTask:667] Received certificate with Id: 1aa9035cc63340a9fe8db7610da4eee0c0e3aa1713a8371e7eb8de754d8f8bef
```

Reply:
I guess we might as well stay consistent. Those log messages are quite verbose though.